pkg/volume/vsphere_volume/vsphere_volume_util.go

```diff
@@ -104,6 +104,7 @@ func (util *VsphereDiskUtil) CreateVolume(v *vsphereVolumeProvisioner) (volSpec
 		Name:       name,
 	}
 
+	volumeOptions.Zone = selectedZone
```

Comment:
So if I understand this correctly, this feature does not uses `NodeAffinity` field of PVs and hence does not uses topology aware provisioning?

Reply:
Yes. Your understanding is correct. The scope of this fix is limited to honouring the allowedTopologies zones during volume provisioning.

Reply:
So if I don't specify `allowedTopology` in my default storageClass and my nodes are spread across zones then it will basically result in PVCs which can not be used by pods.  Previously - we blocked/errored out on volume provisioning altogether if datastore being used is not shared with all VMs (https://github.com/kubernetes/kubernetes/issues/72497).  How does this interact with that bug?

Reply:
The issue in https://github.com/kubernetes/kubernetes/issues/72497 is that volume provisioning will fail if there is no shared datastore available across all kubernetes node VMs. Now, with this change, by specifying allowedTopologies in SC, the volume provisioning can happen as long as there is a shared datastore available for the nodes within the zone. 
There is no change in behaviour when allowedTopology is not specified. It will continue to work in the same way today, i.e by looking for shared datastore across all nodes and succeed or fail as the case is.